### ISSUE TITLE
Uncaught TypeError: Cannot read property 'then' of undefined

### ISSUE THREAD

**Issue by nunoz** [atom/settings-view#879]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.12.1
**Electron Version**: 1.3.6
**System**: Mac OS X 10.12.1
**Thrown From**: [settings-view](https://github.com/atom/settings-view) package, v0.243.1

### Stack Trace

Uncaught TypeError: Cannot read property 'then' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/updates-panel.js:193

TypeError: Cannot read property 'then' of undefined
    at UpdatesPanel.module.exports.UpdatesPanel.updateAll (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/updates-panel.js:193:42)
    at HTMLButtonElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/updates-panel.js:69:24)
    at HTMLButtonElement.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4435:9)
    at HTMLButtonElement.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
     -4:02.7.0 tree-view:remove-project-folder (span.name.icon.icon-file-directory)
     -3:42.4.0 core:paste (atom-text-editor.editor.is-focused)
     -3:20.1.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:11.9.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-atom-dark-syntax.theme-atom-dark-ui)
```

### Config

```json
{
  "core": {
    "telemetryConsent": "no",
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.29.11 (inactive)
atom-sql-list, v0.3.1 (inactive)
atom-typescript, v10.1.6 (active)
autocomplete-json, v5.1.0 (active)
find-selection, v0.16.0 (inactive)
json-schema, v0.1.15 (active)
jumpy, v3.1.1 (inactive)
last-cursor-position, v0.9.0 (active)
linter, v1.11.16 (active)
linter-jshint, v2.1.0 (active)
linter-tslint, v0.11.1 (active)
php-debug, v0.2.3 (active)
pigments, v0.34.0 (active)
project-manager, v3.2.1 (active)
Sublime-Style-Column-Selection, v1.7.2 (active)
todo-show, v1.7.0 (inactive)
atom-dark-syntax, v0.27.0 (active)
atom-dark-ui, v0.52.0 (active)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.45.0 (inactive)
base16-tomorrow-dark-theme, v1.3.0 (inactive)
base16-tomorrow-light-theme, v1.3.0 (inactive)
one-dark-ui, v1.6.2 (inactive)
one-light-ui, v1.6.2 (inactive)
one-dark-syntax, v1.5.0 (inactive)
one-light-syntax, v1.5.0 (inactive)
solarized-dark-syntax, v1.0.5 (inactive)
solarized-light-syntax, v1.0.5 (inactive)
about, v1.7.0 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.13.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.4 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.42.0 (active)
bracket-matcher, v0.82.2 (active)
command-palette, v0.39.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.202.2 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.158.8 (active)
metrics, v1.1.1 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.1 (inactive)
settings-view, v0.243.1 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.4 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.2 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.0 (active)
timecop, v0.33.2 (active)
tree-view, v0.210.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.38.2 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.0 (active)
language-csharp, v0.12.1 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.46.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by 50Wliu** [comment]
Hi @nunoz, what were you doing when this occurred?

**Comment by nunoz** [comment]
Hi,
i was updating installed packages. 
it updated several successfully but 2 missed. So i retried and got that exception.
i restart atom and everything seems ok. I checked if had any available updates and had none.

**Comment by ttsugriy** [comment]
got the same exception after clicking "Update All" button inside "Updates" tab. This happened after I manually clicked "Update" for each individual updates and then, after there are no more available updates, clicked on "Update All".

**Comment by Ben3eeE** [comment]
@damieng @ungb I have reported this behavior previously in https://github.com/atom/settings-view/issues/766#issuecomment-215431294. The update all button updates all packages even if individual packages are updated/pending update already.

**Comment by Ben3eeE** [comment]
I even recorded a gif back then, this is not recent, I stopped doing this since it didn't work 😆 

![update then update all](https://cloud.githubusercontent.com/assets/1058982/20232555/79a89b48-a869-11e6-8818-cb12c15ead2a.gif)

**Comment by damieng** [comment]
This looks like it should be easy to fix, I'm on it.

**Comment by anderso** [comment]
I got an exception and a link to this issue, stating that it was already reported. I had three pending updates in the Updates view. From there I first chose to delete one of them, then clicked Update All, that is when the error message appeared. 

I could try to reproduce but I would need to downgrade some packages and I'm not sure how.

**Comment by joakin** [comment]
Same as @anderso for me. Checked the updates tab, uninstalled an extension before updating anything, then clicked *Update all* and got this exception.

**Comment by damieng** [comment]
Managed to get a stack trace of this and am investigating.

**Comment by vanm** [comment]
Could be useful context: I received this error after uninstalling a package that had a pending update from this page and then clicking "Update All"

**Comment by lee-dohm** [comment]
I just ran into this. Here are my steps to reproduce:

1. Have more than one package with an available update
1. Launch Atom
1. Open package update view by clicking the available updates notification in the status bar
1. Click the Uninstall button on one of the packages
1. Click Update All

**Expected:** Updates to begin and complete successfully
**Actual:** Unhandled exception (see details below)

-----

**Atom Version**: 1.14.0-dev-9fb2f456b
**Electron Version**: 1.3.13
**System**: Mac OS X 10.12.2
**Thrown From**: [settings-view](https://github.com/atom/settings-view) package, v0.244.0


### Stack Trace

Uncaught TypeError: Cannot read property 'then' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/updates-panel.js:193

TypeError: Cannot read property 'then' of undefined
    at UpdatesPanel.module.exports.UpdatesPanel.updateAll (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/updates-panel.js:193:42)
    at HTMLButtonElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/settings-view/lib/updates-panel.js:69:24)
    at HTMLButtonElement.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4435:9)
    at HTMLButtonElement.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
     -0:15.7.0 core:backspace (input.hidden-input)
     -0:14.9.0 core:save (input.hidden-input)
     -0:14.3.0 core:close (input.hidden-input)
     -0:11.6.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-light-syntax.theme-one-light-ui)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "customFileTypes": {
      "source.git-config": [
        "gitconfig"
      ],
      "source.ini": [
        ".flowconfig"
      ],
      "source.ruby": [
        "Brewfile"
      ],
      "text.xml.plist": [
        "plist"
      ]
    },
    "disabledPackages": [
      "background-tips"
    ],
    "followSymlinks": true,
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      ".keep",
      "._*",
      "*.pyc",
      "Thumbs.db"
    ],
    "openEmptyEditorOnStart": false,
    "projectHome": "/Users/Lee/Source",
    "telemetryConsent": "limited",
    "themes": [
      "one-light-ui",
      "one-light-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
advanced-open-file, v0.16.4 (inactive)
atom-2048, v3.1.0 (active)
atom-material-syntax, v0.4.6 (inactive)
atom-material-ui, v1.3.8 (inactive)
atom-package-manager, v1.9.3 (active)
autocomplete-emojis, v2.5.0 (active)
badges, v2.2.4 (active)
block-travel, v1.0.5 (active)
bug-report, v0.7.1 (active)
convert-date, v0.1.1 (inactive)
docblockr, v0.9.1 (active)
double-tag, v0.4.0 (active)
file-type-icons, v1.3.3 (active)
github, v0.0.0 (active)
hex, v0.6.2 (active)
highlight-selected, v0.11.2 (active)
indentation-indicator, v1.0.0 (active)
language-elixir, v0.16.0 (active)
language-generic-config, v1.3.0 (active)
language-pegjs, v0.5.0 (active)
language-rust, v0.4.8 (active)
less-than-slash, v0.16.0 (active)
markdown-helper, v0.2.1 (inactive)
minimap, v4.26.4 (active)
package-sync, v1.1.0 (active)
Pompomodoro, v1.0.0 (active)
pretty-json, v1.6.1 (active)
red-wavy-underline, v0.3.2 (active)
right-case, v0.2.0 (inactive)
set-syntax, v0.3.2 (active)
soft-wrap-indicator, v1.0.1 (active)
sort-lines, v0.14.0 (inactive)
tabs-to-spaces, v1.0.2 (active)
toggle-quotes, v1.0.0 (inactive)
unity-ui, v2.1.7 (inactive)
atom-dark-syntax, v0.28.0 (inactive)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.9.0 (inactive)
one-light-ui, v1.9.0 (active)
one-dark-syntax, v1.6.0 (inactive)
one-light-syntax, v1.6.0 (active)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (inactive)
about, v1.7.2 (active)
archive-view, v0.62.2 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.14.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.34.2 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.29.0 (inactive)
autosave, v0.24.0 (active)
background-tips, v0.26.1 (inactive)
bookmarks, v0.43.4 (active)
bracket-matcher, v0.85.0 (active)
command-palette, v0.39.2 (inactive)
deprecation-cop, v0.55.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.1 (active)
find-and-replace, v0.204.8 (active)
fuzzy-finder, v1.4.1 (active)
git-diff, v1.2.0 (active)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.1 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.159.1 (active)
metrics, v1.1.2 (active)
notifications, v0.65.2 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (active)
snippets, v1.0.4 (active)
spell-check, v0.70.0 (active)
status-bar, v1.7.0 (active)
styleguide, v0.48.0 (active)
symbols-view, v0.113.2 (inactive)
tabs, v0.103.1 (active)
timecop, v0.33.2 (active)
tree-view, v0.211.1 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.2 (active)
whitespace, v0.36.0 (active)
wrap-guide, v0.39.0 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.2 (active)
language-csharp, v0.13.0 (active)
language-css, v0.41.0 (active)
language-gfm, v0.88.0 (active)
language-git, v0.19.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.124.0 (active)
language-json, v0.18.3 (active)
language-less, v0.30.0 (active)
language-make, v0.22.3 (active)
language-mustache, v0.13.1 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.9.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.3 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.1 (active)
language-shellscript, v0.24.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.1 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.13 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by 50Wliu** [comment]
The expected behavior in this case would be to uninstall the package and update all the other ones, correct?

**Comment by lee-dohm** [comment]
Yes, that's correct. I should have been more specific that I had waited for the uninstall to finish before I clicked the Update All button.

**Comment by 50Wliu** [comment]
Going to keep this open, not sure if my PR fixed it or not.

**Comment by parasyte** [comment]
FWIW, I just hit this bug. Echoing STR as provided by @anderso @joakin @damieng @lee-dohm ; uninstall a package, then press the "Update all packages" button.

**Comment by montogeek** [comment]
Same here in Atom 1.13.0

**Comment by razzul** [comment]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.13.0
**Electron Version**: 1.3.13
**System**: Microsoft Windows 7 Professional 
**Thrown From**: [settings-view](https://github.com/atom/settings-view) package, v0.244.0


### Stack Trace

Uncaught TypeError: Cannot read property 'then' of undefined

```
At C:\Users\Administrator\AppData\Local\atom\app-1.13.0\resources\app.asar\node_modules\settings-view\lib\updates-panel.js:193

TypeError: Cannot read property 'then' of undefined
    at UpdatesPanel.module.exports.UpdatesPanel.updateAll (C:\Users\Administrator\AppData\Local\atom\app-1.13.0\resources\app.asar\node_modules\settings-view\lib\updates-panel.js:193:42)
    at HTMLButtonElement.<anonymous> (C:\Users\Administrator\AppData\Local\atom\app-1.13.0\resources\app.asar\node_modules\settings-view\lib\updates-panel.js:69:24)
    at HTMLButtonElement.dispatch (C:\Users\Administrator\AppData\Local\atom\app-1.13.0\resources\app.asar\node_modules\jquery\dist\jquery.js:4435:9)
    at HTMLButtonElement.elemData.handle (C:\Users\Administrator\AppData\Local\atom\app-1.13.0\resources\app.asar\node_modules\jquery\dist\jquery.js:4121:28)
```

### Commands

```
     -0:07.5.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.compact-tab-bar.use-animations.compact-tree-view.theme-atom-dark-syntax.theme-atom-material-ui)
```

### Config

```json
{
  "core": {
    "telemetryConsent": "limited",
    "themes": [
      "atom-material-ui",
      "atom-dark-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-material-syntax, v1.0.2 (inactive)
atom-material-ui, v1.3.9 (active)
git-plus, v7.1.1 (active)
linter, v1.11.21 (active)
linter-php, v1.3.1 (active)
minimap, v4.26.7 (active)
minimap-git-diff, v4.3.1 (active)
atom-dark-syntax, v0.28.0 (active)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.8.2 (inactive)
one-light-ui, v1.8.2 (inactive)
one-dark-syntax, v1.6.0 (inactive)
one-light-syntax, v1.6.0 (inactive)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (inactive)
about, v1.7.2 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.14.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.33.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.29.0 (inactive)
autosave, v0.23.2 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.43.2 (active)
bracket-matcher, v0.85.1 (active)
command-palette, v0.39.1 (inactive)
deprecation-cop, v0.55.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.204.5 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.2.0 (active)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.1 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.159.1 (active)
metrics, v1.1.2 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.5 (active)
status-bar, v1.6.0 (active)
styleguide, v0.48.0 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.1 (active)
timecop, v0.33.2 (active)
tree-view, v0.211.1 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.39.0 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.1 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by sliceofcode** [comment]
1. Click "Update All"

**Atom Version**: 1.13.1
**Electron Version**: 1.4.15
**System**: linux 4.9.6-1-ARCH
**Thrown From**: [settings-view](https://github.com/atom/settings-view) package, v0.244.0


### Stack Trace

Uncaught TypeError: Cannot read property 'then' of undefined

```
At /usr/lib/atom/node_modules/settings-view/lib/updates-panel.js:193

TypeError: Cannot read property 'then' of undefined
    at UpdatesPanel.module.exports.UpdatesPanel.updateAll (/usr/lib/atom/node_modules/settings-view/lib/updates-panel.js:193:42)
    at HTMLButtonElement.<anonymous> (/usr/lib/atom/node_modules/settings-view/lib/updates-panel.js:69:24)
    at HTMLButtonElement.dispatch (/usr/lib/atom/node_modules/jquery/dist/jquery.js:4435:9)
    at HTMLButtonElement.elemData.handle (/usr/lib/atom/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
     -2:09.7.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-always.theme-ariake-dark-syntax.theme-one-dark-ui)
     -1:54.1.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.theme-ariake-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "autoHideMenuBar": true,
    "automaticallyUpdate": false,
    "disabledPackages": [
      "autocomplete-html",
      "autocomplete-snippets",
      "background-tips",
      "bookmarks",
      "encoding-selector",
      "git-diff",
      "go-to-line",
      "language-clojure",
      "link",
      "styleguide",
      "spell-check",
      "welcome",
      "archive-view",
      "autocomplete-atom-api",
      "autocomplete-css",
      "autocomplete-plus",
      "snippets"
    ],
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "._*",
      "Thumbs.db",
      "node_modules",
      "bower_components"
    ],
    "projectHome": "/home/cheesy/dev/prj",
    "reopenProjectMenuCount": 30,
    "telemetryConsent": "limited",
    "themes": [
      "one-dark-ui",
      "ariake-dark-syntax"
    ],
    "warnOnLargeFileLimit": 10
  }
}
```

### Installed Packages

```coffee
# User
ariake-dark-syntax, v0.1.3 (active)
duotone-dark-sea-syntax, v2.1.0 (inactive)
ex-mode, v0.13.1 (inactive)
file-icons, v2.0.9 (active)
language-babel, v2.49.4 (active)
language-lua, v0.9.11 (active)
language-moonscript, v1.7.1 (active)
language-rust, v0.4.9 (active)
linter, v1.11.21 (active)
linter-rust, v0.7.2 (active)
vim-mode-plus, v0.80.0 (active)
vim-mode-plus-ex-mode, v0.7.1 (inactive)
atom-dark-syntax, v0.28.0 (inactive)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.8.2 (active)
one-light-ui, v1.8.2 (inactive)
one-dark-syntax, v1.6.0 (inactive)
one-light-syntax, v1.6.0 (inactive)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (inactive)
about, v1.7.2 (active)
archive-view, v0.62.0 (inactive)
autocomplete-atom-api, v0.10.0 (inactive)
autocomplete-css, v0.14.1 (inactive)
autocomplete-html, v0.7.2 (inactive)
autocomplete-plus, v2.33.1 (inactive)
autocomplete-snippets, v1.11.0 (inactive)
autoflow, v0.29.0 (inactive)
autosave, v0.23.2 (active)
background-tips, v0.26.1 (inactive)
bookmarks, v0.43.2 (inactive)
bracket-matcher, v0.85.1 (active)
command-palette, v0.39.1 (inactive)
deprecation-cop, v0.55.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (inactive)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.204.5 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.2.0 (inactive)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.1 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.159.3 (active)
metrics, v1.1.3 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (active)
snippets, v1.0.4 (inactive)
spell-check, v0.68.5 (inactive)
status-bar, v1.6.0 (active)
styleguide, v0.48.0 (inactive)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.1 (active)
timecop, v0.33.2 (active)
tree-view, v0.211.1 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (inactive)
whitespace, v0.35.0 (active)
wrap-guide, v0.39.0 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (inactive)
language-coffee-script, v0.48.1 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.47.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by sliceofcode** [comment]
Note: After clicking on "Update All", it grayed out. Then I clicked on "Check for Updates", and most of the packages that were listed as having updates disappeared. I can only assume that they got updated somehow. There were 2 packages left in the list. This time, clicking "Update All" did its job properly, and after restarting Atom after updating leaves all of my packages updated.

**Comment by 50Wliu** [comment]
@sliceofcode Thanks for reaching out!

We've determined that you are using an unofficial build or distribution of Atom. Often these customized versions of Atom are modified versions of the Stable branch of Atom with mismatched versions of built-in components. These updated components are taken from the Beta channel or `master` branch and then injected into the Stable version and a new Atom package is generated. Because of the way Atom is constructed, using these mismatched components can cause mysterious and hard-to-diagnose problems. You can find out more about [why we chose to not support unofficial distributions here](https://github.com/atom/design-decisions/blob/master/unofficial-atom-distributions.md).

**Comment by diggit** [comment]
My scenario to get this error:
1. click on notification about available updates (when there are any) in bottom right corner, list of packages to update appears
2. uninstall some package from this list (from this view)
3. click "update all"
4. error pops up

It looks like atom does not expect changes in list of packages to update. "Update all" should probably check for packages being really present in system right before update.

**Comment by 50Wliu** [comment]
Yeah, right now I'm waiting to see if we get any reports on Atom 1.14.0 and later before closing it.

**Comment by sliceofcode** [comment]
@50Wliu Huh, okay. Does [atom-editor-beta-bin from the AUR](https://aur.archlinux.org/packages/atom-editor-beta-bin) count as an unofficial distribution? From reading the PKGBUILD, all it does is grab the `.deb` from `https://atom-installer.github.com` and extract it. The only thing that's patched is the `.desktop` file, does that make it unofficial? The package's description says "Chrome-based text editor from Github - Beta Channel - Precompiled binary from official repository". Maybe something wonky happened, so I reset everything.

**Comment by 50Wliu** [comment]
It's still technically unofficial, but is a much safer bet than whatever distribution you were using previously.

**Comment by maximumdata** [comment]
On 1.14.1, and just received this error after doing the steps mentioned by other users. Clicked the update notification in the status bar, uninstalled a package, and then clicked update all.

**Comment by lparsons** [comment]
I just ran into this bug 1.14.3 (same steps as previously described).

**Comment by bronson** [comment]
Just hit this on 1.15.0.  My actions were the same as yall's: uninstall a package, then "update all".

**Comment by alopix** [comment]
Same problem with 1.15.0
I updated a package and after it installed clicked the update all button, which of course cannot do anything with just this one update available (which was already updated by then).

**Comment by noplanman** [comment]
Still an issue on 1.16.0

Exactly as described by @alopix in the previous comment.

**Comment by scottaj91** [comment]
Still an issue in 1.17.2. Same steps as everyone else.

**Comment by davisonio** [comment]
Yep I'm having this is 1.17.2 too.

**Comment by deepmehtait** [comment]
I am having same issue on 1.18.0 x64 Mac OS as well.
```
TypeError: Cannot read property 'then' of undefined
at UpdatesPanel.updateAll
(/Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/updates-panel.js:298:37)
at HTMLButtonElement.onclick
(/Applications/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/updates-panel.js:142:36)
```

**Comment by 50Wliu** [comment]
At the bare minimum https://github.com/atom/settings-view/blob/ea7f18e089fcd4fbcb549aa04dd963437e26f2c0/lib/package-card.js#L662 needs to return a Promise.  Though the Updates panel can be more intelligent and not try to update uninstalled/already-updated packages.

**Comment by ioquatix** [comment]
Same problem with v1.19.0

**Comment by storypixel** [comment]
popped up for me with 1.20.0

**Comment by ninthart** [comment]
Yep, popped up for me with 1.20.1 after uninstalling a package from the updates list and then hitting Update All…

**Comment by CaffeinatedAndroid** [comment]
Still not fixed in 1.21.0

Steps to reproduce:
1- Manual update one package
2- Wait for update to complete
3- Click 'Update All'

Note that it works after I close Preferences tab, then re-open Preferences > Updates, then the click Update All button.
